src/exercises/day-3/safe-ffi-wrapper.md

```diff
@@ -10,13 +10,36 @@ You will want to consult the manual pages:
 * [`readdir(3)`](https://man7.org/linux/man-pages/man3/readdir.3.html)
 * [`closedir(3)`](https://man7.org/linux/man-pages/man3/closedir.3.html)
 
-You will also want to browse the [`std::ffi`] module, particular for [`CStr`]
-and [`CString`] types which are used to hold NUL-terminated strings coming from
-C. The [Nomicon] also has a very useful chapter about FFI.
+You will also want to browse the [`std::ffi`] module. There you find a number of
+string types which you need for the exercise:
+
+| Types                      | Encoding       | Use                            |
+|----------------------------|----------------|--------------------------------|
+| [`str`] and [`String`]     | UTF-8          | Text processing in Rust        |
+| [`CStr`] and [`CString`]   | NUL-terminated | Communicating with C functions |
+| [`OsStr`] and [`OsString`] | OS-specific    | Communicating with the OS      |
+
+You will convert between all these types:
+
+- `&str` to `CString`: you need to allocate space for a trailing `\0` character,
+- `CString` to `*const i8`: you need a pointer to call C functions,
+- `*const i8` to `&CStr`: you need something which can find the trailing `\0` character,
```

Comment:
`*const i8` to `&[u8]` might be useful for other encodings/to write your own abstractions

Reply:
Right, definitely and that's what you get if you go `*const i8` to `&CStr` to `&[u8]` which is the next line. I think we agree :smile: